clients/http_json_rpc_client.go

```diff
@@ -91,24 +111,21 @@ func NewGenericHttpJsonRpcClient(
 
 			client.batchRequests = make(map[interface{}]*batchRequest)
 		}
+
 		if jsonRpcCfg.EnableGzip != nil {
 			client.enableGzip = *jsonRpcCfg.EnableGzip
 		}
+
 		if jsonRpcCfg.Headers != nil {
 			client.headers = jsonRpcCfg.Headers
 		}
-	}
 
-	if util.IsTest() {
-		client.httpClient = &http.Client{}
-	} else {
-		client.httpClient = &http.Client{
-			Timeout: 60 * time.Second,
-			Transport: &http.Transport{
-				MaxIdleConns:        1024,
-				MaxIdleConnsPerHost: 256,
-				IdleConnTimeout:     90 * time.Second,
-			},
+		if jsonRpcCfg.ProxyPool != "" {
+			client.proxyPool = jsonRpcCfg.ProxyPool
+		}
+
+		if fetchedProxyPool != nil {
+			client.fetchedProxyPool = fetchedProxyPool
```

Comment:
Do we even need ProxyPool name? Why not just a `proxyPool *ProxyPool` passed object?